packages/flutter/lib/src/rendering/view.dart

```diff
@@ -245,8 +245,16 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
 
   void _updateSystemChrome() {
     final Rect bounds = paintBounds;
-    final Offset top = Offset(bounds.center.dx, _window.padding.top / _window.devicePixelRatio);
-    final Offset bottom = Offset(bounds.center.dx, bounds.center.dy - _window.padding.bottom / _window.devicePixelRatio);
+    // center of the status bar
+    final Offset top = Offset(
+      bounds.center.dx,
+      _window.padding.top / 2.0,
```

Comment:
info: bottom padding shouldn't be divided by `_window.devicePixelRatio` because everywhere we use native pixels. We should just divide the top padding by two so we get the status bar center. 
Before this calculations worked correctly only on 2x screens.